core/shared/src/main/scala-3/zio/ZIOVersionSpecific.scala

```diff
@@ -19,16 +19,30 @@ private[zio] transparent trait ZIOVersionSpecific[-R, +E, +A] { self: ZIO[R, E,
   def provideSome[R0] =
     new ProvideSomePartiallyApplied[R0, R, E, A](self)
 
+  /**
+   * Same as [[provideSome]], but does not require providing remainder
+   *
+   * {{{
+   * val clockLayer: ZLayer[Any, Nothing, Clock] = ???
+   *
+   * val zio: ZIO[Clock with Random, Nothing, Unit] = ???
+   *
+   * val zio2 = zio.provideSome(clockLayer) // Inferred type is ZIO[Random, Nothing, Unit]
+   * }}}
+   */
```

Comment:
I think we should add a note for intelliJ users here and mention that type inference won't work when using the built-in highlighting

Reply:
Fixed - I've added comment with instruction how to enable that